src/generators/genlua.ml

```diff
@@ -578,6 +636,15 @@ and gen_expr ?(local=true) ctx e = begin
 				spr ctx (ident v.v_name);
 			| Some e ->
 				match e.eexpr with
+				| TCall _ when is_multireturn ctx e ->
+				    if Meta.has Meta.MultiReturn v.v_meta then
+					raise (Failure "@:MultiReturn metadata must not be set on variables receiving multiple returns");
+				    v.v_meta <- [(Meta.MultiReturn, multi_value_meta ctx e, null_pos)];
+				    let local_tmp_names = multi_value_tmp_names v.v_meta in
+				    v.v_name <- String.concat ", " local_tmp_names;
+				    print ctx "local %s =" v.v_name;
+				    gen_value ctx e;
+				    semicolon ctx;
```

Comment:
If we do need to create temporary variables, we create some metadata on the TVar that keeps a map between the original field name, and the generated temp name.  This is done by creating a Meta.MultiReturn metadata on the TVar, and then creating an EOBjectDecl that contains a series of name -> EConst(String("name")) tuples.